src/modules/roc_sndio/ibackend.h

```diff
@@ -42,20 +42,19 @@ class IBackend {
         FilterDevice = (1 << 3)
```

Comment:
There used to be a check in Pulseaudio::Backend::probe() which checked to see if the filter_flags corresponded to a FilterSink which now is moved to Pulseaudio::open_sink().Was this check not needed at all in that case?

Reply:
That check guaranteed that we wont select PulseaudioBackend when opening sources (because it supports only sinks). The previous logic was to select backend first, and then call open, so it was important to select right backend.

Now we're just calling open on all backends iteratively. PulseaudioBackend::open_source() will return null, and backend dispatched will go to the next backend, so everything is ok.